engine/verification/assigner/engine.go

```diff
@@ -87,12 +97,162 @@ func (e *Engine) Process(originID flow.Identifier, event interface{}) error {
 }
 
 func (e *Engine) process(originID flow.Identifier, event interface{}) error {
-	return nil
+	return fmt.Errorf("assigner engine is not supposed to invoked by process method")
+}
+
+// handleExecutionReceipt receives a receipt that appears in a finalized container block. In case this verification node
+// is staked at the reference block of this execution receipt's result, chunk assignment is done on the execution result, and
+// the assigned chunks' locators are pushed to the chunks queue, which are made available for the chunk queue consumer (i.e., the
+// fetcher engine).
+func (e *Engine) handleExecutionReceipt(receipt *flow.ExecutionReceipt, containerBlockID flow.Identifier) {
+	receiptID := receipt.ID()
+	resultID := receipt.ExecutionResult.ID()
+	referenceBlockID := receipt.ExecutionResult.BlockID
+
+	log := log.With().
+		Hex("receipt_id", logging.ID(receiptID)).
+		Hex("result_id", logging.ID(resultID)).
+		Hex("reference_block_id", logging.ID(referenceBlockID)).
+		Hex("container_block_id", logging.ID(containerBlockID)).
+		Logger()
+
+	// verification node should be staked at the reference block id.
+	ok, err := e.stakedAtBlockID(referenceBlockID)
+	if err != nil {
+		log.Error().Err(err).Msg("could not verify stake of verification node for result at reference block id")
+		return
+	}
+	if !ok {
+		log.Debug().Msg("node is not staked at reference block id, receipt is discarded")
+		return
+	}
+
+	// chunk assignment
+	chunkList, err := e.chunkAssignments(context.Background(), &receipt.ExecutionResult)
+	if err != nil {
+		log.Error().Err(err).Msg("could not determine chunk assignment")
+		return
+	}
+	log.Info().
+		Int("total_assigned_chunks", len(chunkList)).
+		Msg("chunk assignment done")
+
+	// pushes assigned chunks to
```

Comment:
unfinished comment

Reply:
fixed in https://github.com/onflow/flow-go/pull/427/commits/1633985bce8bfe968cb772832c788e8ef71b0ec8.